src/codeu/chat/server/Controller.java

```diff
@@ -41,17 +34,44 @@ public Controller(Uuid serverId, Model model) {
 
   @Override
   public Message newMessage(Uuid author, Uuid conversation, String body) {
-    return newMessage(createId(), author, conversation, body, Time.now());
+    Uuid id = createId();
+    Time time = Time.now();
+    String messageLog = "ADD-MESSAGE" + SPACE +
```

Comment:
creating the messageLog should be a single Util function. imagine you want to change your log format, now you have to put the change in 3 different places, newMessage(), newUser() and newConversation, instead if you have a TransactionLogUtil.writeLog(type, ...) {} you can have a switch case based on type and have all the logic inside one function. Similarly, reading the log should be TransactionLogUtil.readLog() ... and then your types: "ADD-MESSAGE", "ADD-USER" etc can be constants in this TransactionLogUtil class.